explore/src/main/scala/explore/itc/ItcPanelTitle.scala

```diff
@@ -27,64 +28,67 @@ case class ItcPanelTitle(
   itcPanelProps:   ItcProps,
   itcGraphResults: Pot[ItcAsterismGraphResults],
   tileState:       View[SelectedItcTarget]
-) extends ReactFnProps(ItcPanelTitle.component) {
+) extends ReactFnProps(ItcPanelTitle.component):
   val selectedTarget = tileState.zoom(SelectedItcTarget.value.asLens)
-}
 
 object ItcPanelTitle:
   private type Props = ItcPanelTitle
 
   private val component =
-    ScalaFnComponent[Props] { props =>
-      def newSelected(p: Int): Option[ItcTarget] =
-        props.itcPanelProps.targets.lift(p)
+    ScalaFnComponent
+      .withHooks[Props]
+      // Reset the selected target if it changes
+      .useEffectWhenDepsReadyBy(props => props.itcGraphResults.map(_.brightestTarget)): props =>
```

Comment:
We probably want `useEffectWhenDepsReadyOrChange` if we want it to execute when the brightest target changes.